app/main.py

```diff
@@ -1,3 +1,36 @@
-def shop_trip():
-    # write your code here
-    pass
+from typing import List
+
+import json
+from app.client_resources.Point2D import Point2d
+from app.client_resources.customer import Customer
+from app.shop_resources.shop import Shop
+from app.client_resources.car import Car
+
+
+def create_shops(shops: list[dict]) -> List[Shop]:
+    return [Shop(shop_data["name"],
+                 Point2d(shop_data["location"]),
+                 shop_data["products"]) for shop_data in shops]
+
+
+def process_customers_list() -> List[Customer]:
+    try:
+        with open("app/config.json", "r") as f:
+            world_data = json.load(f)
+    except FileNotFoundError as fnfe:
+        print(f"Loading problem, {fnfe} doesn't exist.")
+        raise fnfe
+
+    return [Customer(name=customer["name"],
+                     product_cart=customer["product_cart"],
+                     location=Point2d(customer["location"]),
+                     money=customer["money"],
+                     car=Car(**customer["car"]),
+                     known_shops=create_shops(world_data["shops"]))
+            for customer in world_data["customers"]]
+
+
+def shop_trip() -> None:
+    customers = process_customers_list()
+    for customer in customers:
+        customer.wake_up_and_ride()
```

Comment:
On every iteration, you call the method wake_up_and_ride which calls the method check_fuel_price which opens a file and gets the same data every time. It takes extra time, try to get all data with only one file opening.

Reply:
Was pretty sure that program logic is to get exact fuel price by the time we getting the ride costs(I mean the data could be changed and that `json` is just hypothetical), so yup, it should've been checked every time when we're going to get a ride price... Idk which logic makes same price for every customer, on every ride. If it's like this, we can just simply make constants and get rid of file. I really can't get your idea. We _should_ check it every time, no?